code-challenges.js

```diff
@@ -10,8 +10,39 @@ var people = [
 ]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is a president of the galaxy." "Arthus Dent is a radio employee."
 
+// Declare a function that takes in an array of objects
+const giveInfo = (array) => {
+  // I need to iterate through all of the objects and perform the same fuction. I will prbably use a .map, which will eventaully store strings. For my final return, I plan to reiterate through the array created by this .map and return each string.
+  let arrayOfFixedNamedObjects = array.map(value => {
+    // I will craete a deep copy of the object so that the original input is not altered
+    let copyObject = JSON.parse(JSON.stringify(value));
+    // Inside of my map, I need to isolate the name first, because I need to capitalize the first and last names.To isolate the names. I'm going to do a .split on the string and store the newly created arry into a new variable
 
+    let charArray = copyObject.name.split("");
+    // Then I need to capitalize the first letter of the array.
+    charArray[0] = charArray[0].toUpperCase();
+    
+    // Then I need to find where the first space is and capitalize the character immedately after it
+    charArray[charArray.indexOf(" ") + 1] = charArray[charArray.indexOf(" ") + 1].toUpperCase();
+    // Then I will join this updated array into a new string name to be called upon later. 
+    let updatedName = charArray.join("");
+    // The new name will be replace the old name of the copied object
+    copyObject.name = updatedName;
+    
+    // Return a sting of the final output
+    return `${copyObject.name} is a ${copyObject.occupation}.`;
+  }) // This will end the .map
 
+  // Now I just need to iterate through the array generated by the proevious .map and return the sentences one-by-one
+  for (let i = 0; i < arrayOfFixedNamedObjects.length; i++) {
+    console.log(arrayOfFixedNamedObjects[i]);
+  }
+  //Ends the for loop
+} // Ends function
+
+// Test to make sure that the function works
+console.log("-----CHALLENGE 1-----");
+giveInfo(people);
```

Comment:
Great work here, but I would want to remind you that we should not be using console.logs inside our functions. If this was to be used in our live environments, or in production, the function would never output anything.  Instead we should be using returns and then when we call our function, we use a console.log. 

I understand if you wanted to try it here, in the assessment, but if we take this as we should, a live practice for what we can expect in a real world scenario- we would want to our pull-request to be out 'ready for live, ready to merge' code to be reviewed. 

In this situation, the code would have been rejected to be refactored, thus costing time/money. 

Just keep this in mind!